routing/fallbackgroup.go

```diff
@@ -5,51 +5,16 @@ package routing
 
 // currently only for load balanced routes
 func applyFallbackGroups(r []*Route) []*Route {
-	groups := make(map[string][]*Route)
 	for i := range r {
-		for j := range r[i].Predicates {
-			er := r[i].Route
-
-			// NOTE: this one "LBMember" is now hard coded to omit circular dependency.
-			// It can be fixed if we implement this logic as a post processor.
-			if er.Predicates[j].Name != "LBMember" {
-				continue
-			}
-
-			if len(er.Predicates[j].Args) == 0 {
-				continue
-			}
-
-			name, ok := er.Predicates[j].Args[0].(string)
-			if !ok {
-				continue
-			}
-
-			groups[name] = append(groups[name], r[i])
-		}
-	}
-
-	for name, group := range groups {
-		if len(group) == 0 {
+		l := len(r[i].Filters)
+		if l < 1 {
 			continue
 		}
 
-		// TODO: here we could clean off load balancing, if there's only a single route
-
-		head := group[0]
-		head.Head = head
-		head.Me = head
-		head.Group = name
-		head.IsLoadBalanced = true
-
-		current := head
-		for _, route := range group[1:] {
-			current.Next = route
-			current = route
-			current.Head = head
-			current.Me = current
-			current.Group = name
-			current.IsLoadBalanced = true
+		f := r[i].Filters[l-1]
```

Comment:
see my comment in the proxy.go, the same applies here, too

Reply:
yes, but only if it's at route parse time and not in the hot path